File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
    int wordL = 1;
    int maxWord = 0;
    map<char,int> letters;
    int a = 0;
    int b = 1;
    letters[s[a]] = 1;
    while (a < s.size()) {
        char letter  = s[b];
        if (letters[letter] == 1 || b == s.size()) {
            if (maxWord < wordL) maxWord = wordL;
            a++;
            wordL = 1;
            b = a + 1;
            letters = {};
            letters[s[a]] = 1;
        }
        else {
            letters[letter] = 1;
            wordL += 1;
            b++;
        }
    }
    return maxWord;
}





};

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        setWindow = set()
        i = 0
        currentLength = 0
        length = 0
        for j in range(len(s)):
            while s[j] in setWindow:
                setWindow.remove(s[i])
                currentLength -= 1
                i += 1
            setWindow.add(s[j])
            currentLength += 1
            length = max(length, currentLength)
        return length